### DISCUSSION TITLE
Immich over Tailscale the easy way

### DISCUSSION THREAD

**Discussion by VomitFartz** [immich-app/immich#4255] [General]
I'm not sure where to post this as I'm new to github and being social online, but I made a write up on how I'm using Immich with Tailscale on ios and thought it might be useful to someone. 

USING IMMICH WITH TAILSCALE THE EASY WAY.

	I'll start out by explaining my set up (just the basics). I have an older 2nd gen i3 laptop
	that I have installed UBUNTU 22.04 on, past that I haven't done much just installed docker, curl, openssh and tailscale. Immich was installed via the curl script and I installed tailscale the same. There are two phones that I was wanting to use with tailscale, an IPhone 7 (my old phone) and an IPhone 8 (my current phone). The IPhone 7 I keep connected to my wifi as it no longer has service, but my IPhone 8 almost never connects to my home wifi, but because I will sometimes use it to connect to jellyfin when I'm out and about I have installed tailscale on it which allows me to connect simply.

	Now, while messing with Immich and backing up all the photos on my old phone (minus a few with weird libraryid values) I wanted to start uploading photos from my current phone. I don't keep wifi turned on, on that phone so I figured Tailscale would be the key. I installed the Immich app for ios turned on taiscale and tried connecting to it by giving the server address and port I used before (on the older phone) 192.168.1.xx:2283. This did not work! So I took to the internet to figure out to make it work. I found one Reddit post where the solution seemed a bit more in depth and a post on the Umbrel community where they had asked the how to do this, but there was no reply. 

	I had logged into the Tailscale admin panel and was going to try to start setting up subnet routes when I decided to first try the ip that Tailscale assigned my laptop running Immich.
	So back on my IPhone 8 I turned on Tailscale, and opened the Immich app. This time instead of 192.68.1.xx:2283 I used the Tailscale ip of that server 100.xx.xx.xx:2283.

	I am now using Immich over Tailscale because "It Just Works".

						Hope this helps someone.
								-VomitFartz

**Comment by avinal** [comment]
Nice one @VomitFartz, I am using tailscale with raspberry pi and yes it just works.

**Comment by luboszima** [comment]
Somehow in 2024 it's not working. 

I am trying to do the same steps but without success.

1) Download iphone app - Immich
2) Synology NAS running immich with tailscale
3) mobile app, set there IP of NAS 2283 port - login and nothing
3a) try same IP with login in browser on my phone -> working.

Tailscale is setup also on my phone, from browser I am able to login. So thinking was in iOS some system change that App can't access VPN IPs?

**Reply by avinal** [reply]
I do not own an iPhone but I have an iPad. It is working fine with tailscale IP but not with tailscale domain.

**Reply by luboszima** [reply]
So I tried it on iPhone, iPad, on my Macbook (when I downloaded app from App Store) but ... with native app I don't have luck.
<img width="444" alt="image" src="https://github.com/user-attachments/assets/bc00abce-69d6-44fd-ae64-9205c9931978">

I tried, Tailscale IP, Local IP and still the same error.
I tried also create new user with simple password ... and also without success.

I am aware if I need enable something on server or do something in mobile phone? (yeah I checked already settings on both devices). :(

**Reply by luboszima** [reply]
I checked Logs and it looks http://<ip>:<port>/api page not exist. So if there is some extra service which needs to be created ... it can be the issue.

**Reply by luboszima** [reply]
I used this docker-compose: https://github.com/immich-app/immich/blob/main/docker/docker-compose.yml

**Reply by luboszima** [reply]
This is what I get from access to /api from browser: 
<img width="489" alt="image" src="https://github.com/user-attachments/assets/80240198-d3e5-4e5d-8bb4-52b9695706dd">

**Reply by avinal** [reply]
You don't need to put /api at the last. That was a requirement in earlier versions. Did you try that? Additionally can you please share your docker-compose.yaml, if possible.

**Reply by avinal** [reply]
Here is the config from my iPad, I am using the tailscale assigned IP with a different port than default. 
![PXL_20240806_164726707~2.jpg](https://github.com/user-attachments/assets/3ed89e4b-96f1-4af8-ba62-10e40a17890a)

**Reply by luboszima** [reply]
I did it before too without /api but same result.

Thank you for your info. I'll check mine.

These logs are after changing url 
<img width="1024" alt="image" src="https://github.com/user-attachments/assets/80343527-359b-417e-bc68-9440d1f8126f">

<img width="256" alt="image" src="https://github.com/user-attachments/assets/b7a4d4f3-c34b-44d6-b77d-cd905575e69a">
<img width="597" alt="image" src="https://github.com/user-attachments/assets/5031f815-4c28-4cde-9acb-1b006ae438cf">

I am trying to dig in codebase of Immich to understand what happening. 
My server version is https://github.com/immich-app/immich/blob/v1.109.2

btw: Thank you for helping me 🙏

**Reply by luboszima** [reply]
Ok I updated to newest version ... and it's working. 
<img width="259" alt="image" src="https://github.com/user-attachments/assets/35b16ce5-0c79-4ae2-b80a-430aa2bf227c">

So whole drama because of outdated version of Immich.

**Reply by avinal** [reply]
Ah yes, that can happen until they reach stability, I think its 3rd or 4th big breaking change that was introduced recently. Glad to know that it is working now 🙏.
